src/main/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserController.java

```diff
@@ -78,14 +82,71 @@
                                                      @RequestParam(value = "size", required = false) Integer size) {
 
-        ResponseEntity<?> profUsersEntityResponse;
         log.info("ProfessionalExternalUserController::findUsersByOrganisation:" + organisationIdentifier);
-        profExtUsrReqValidator.validateRequest(organisationIdentifier, showDeleted, email, status);
+        profExtUsrReqValidator.validateRequest(organisationIdentifier, showDeleted, status);
+        ServiceAndUserDetails serviceAndUserDetails = (ServiceAndUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        boolean isRolePuiUserManager = organisationIdentifierValidatorImpl.ifUserRoleExists(serviceAndUserDetails.getAuthorities(), "pui-user-manager");
+        ResponseEntity<?> profUsersEntityResponse;
+
+        if (!isRolePuiUserManager) {
+            if (StringUtils.isEmpty(status)) {
+                status = "Active";
+            }
+            profExtUsrReqValidator.validateStatusIsActive(status);
+        }
+
+        profUsersEntityResponse = searchUsersByOrganisation(organisationIdentifier, showDeleted, true, status, page, size);
+
+        return profUsersEntityResponse;
+    }
+
+    @ApiOperation(
+            value = "Retrieves the user with the given email address if organisation is active or showDeleted flag ",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            }
+    )
+    @ApiParam(
+            name = "email",
+            type = "string",
+            value = "The email of the desired user to be retrieved",
+            required = false
+    )
+    @ApiResponses({
+            @ApiResponse(
+                    code = 200,
+                    message = "A professional user along with their details",
+                    response = ProfessionalUsersEntityResponse.class
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "An invalid email was provided"
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Invalid authorization"
+            ),
+            @ApiResponse(
+                    code = 404,
+                    message = "No user was found with the provided email address"
+            )
+    })
+    @GetMapping(
+            value = "/user",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE
+    )
+    @Secured({"pui-user-manager"})
+    public ResponseEntity<ProfessionalUsersResponse> findUserByEmail(@ApiParam(hidden = true) @OrgId String organisationIdentifier,
+                                                                     @ApiParam(name = "email", required = false) @RequestParam(value = "email", required = false) String email) {
+
+        ResponseEntity<ProfessionalUsersResponse> profUsersEntityResponse = null;
```

Comment:
Please remove the hard-coded null reference. I realise you will get a compilation error initially and that you are only having a null reference returned if you cannot find the email address. However this is not a great pattern as it could lead to a null pointer. Please wrap any potential null reference in an Optional<String>. And you then need to explicitly decide what to de with an empty email? Is this an exceptional case, etc. If so, you should actively check for the nullness and return an exception response instead.

Reply:
Yes this was going to be tackled, for the sake of seeing how the tests would be affected by the splitting of the endpoint - but I will use an Optional, thanks for the review 😄